test/unix/test_lwt_bytes.ml

```diff
@@ -496,4 +497,32 @@ let suite = suite "lwt_bytes" [
       let check = "abcaaa" = Lwt_bytes.to_string buf in
       Lwt.return check
     end;
+
+    test "bytes read" begin fun () ->
+      let test_fd = "bytes_io_data.ml" in
+      let unix_fd = Unix.openfile test_fd [O_RDONLY] 0 in
+      let fd = Lwt_unix.of_unix_file_descr unix_fd in
+      let buf = Lwt_bytes.create 6 in
+      Lwt_bytes.read fd buf 0 6
+      >>= fun _n ->
+      let check = "abcdef" = Lwt_bytes.to_string buf in
+      Lwt.return check
+    end;
+
+    test "bytes write" begin fun () ->
+      let test_fd = "bytes_io_data_write.ml" in
+      let unix_fd = Unix.openfile test_fd [O_RDWR;O_TRUNC; O_CREAT] 0o666 in
```

Comment:
> I had to generate a fd from the Unix module and transform it to an Lwt file descriptor. In the second test, I had to close the underlying Unix fd that I used to create the Lwt fd in order to flush everything before reading what has been written for the test. Should I close all the fd using the Unix.close at the end of the tests? I have the feeling that it is not really clean.

I recommend using `Lwt_unix.openfile` and `Lwt_unix.close`. I think we can rely, in these tests, on `Lwt_unix` tests having already run.

Reply:
Yes, I think so.